third-party/realdds/src/dds-stream-server.cpp

```diff
@@ -40,32 +30,40 @@ dds_stream_server::~dds_stream_server()
 }
 
 
-dds_video_stream_server::dds_video_stream_server( std::string const & name,
-                                                  dds_stream_profiles const & profiles,
-                                                  int default_profile_index )
-    : dds_stream_server( name, profiles, default_profile_index )
+dds_video_stream_server::dds_video_stream_server( std::string const& stream_name, std::string const& sensor_name )
+    : dds_stream_server( stream_name, sensor_name )
 {
-    for( auto & profile : profiles )
-        if( ! std::dynamic_pointer_cast< dds_video_stream_profile >( profile ) )
-            DDS_THROW( runtime_error, "profile '" + profile->to_string() + "' is not a video profile" );
 }
 
 
-dds_motion_stream_server::dds_motion_stream_server( std::string const & name,
-                                                    dds_stream_profiles const & profiles,
-                                                    int default_profile_index )
-    : dds_stream_server( name, profiles, default_profile_index )
+void dds_video_stream_server::check_profile( std::shared_ptr< dds_stream_profile > const & profile ) const
 {
-    for( auto & profile : profiles )
-        if( ! std::dynamic_pointer_cast< dds_motion_stream_profile >( profile ) )
-            DDS_THROW( runtime_error, "profile '" + profile->to_string() + "' is not a motion profile" );
+    super::check_profile( profile );
+    if( ! std::dynamic_pointer_cast< dds_video_stream_profile >( profile ) )
+        DDS_THROW( runtime_error, "profile '" + profile->to_string() + "' is not a video profile" );
+}
+
+
+dds_motion_stream_server::dds_motion_stream_server( std::string const & stream_name, std::string const & sensor_name )
+    : dds_stream_server( stream_name, sensor_name )
+{
+}
+
+
+void dds_motion_stream_server::check_profile( std::shared_ptr< dds_stream_profile > const & profile ) const
+{
+    super::check_profile( profile );
+    if( ! std::dynamic_pointer_cast< dds_motion_stream_profile >( profile ) )
```

Comment:
I think we need a way to check if a `dds_stream_profile` is `dds_video_stream_profile` (or ``dds_motion_stream_profile`). Maybe making `check_profile` or a similar function that uses `dynamic_cast` public.

Reply:
Don't understand the suggestion.
For now, this is good enough, I think.

Reply:
In the client, I sometimes have to call different functions depending on the profile type (e.g. `software_sensor::add_video_stream` or `software_sensor::add_motion_stream`), and profiled are mostly passed as the base `dds_stream_profile`.

For now I managed, but I feel we might need to add in the near future a method to verify the actual type.